app/models/tag.rb

```diff
@@ -51,7 +56,9 @@ def display_name()
   # Strips the tag's name and returns the color details if present
   # if no color information is found, returns a default value of #ccc
   def color()
-    name[/\A(!\h{6})_[[:word:]]+?\z/,1].try(:gsub, "!", "#") || "#555"
+    return '' if self.name.nil?
```

Comment:
ditto here, how do we get to this situation, and can it be prevented w/ ActiveRecord validation?

Reply:
Same reason as `#display_name`

I have refactored my implementation